src/charm.py

```diff
@@ -258,11 +262,19 @@ def _update_workload_container_config(self, event):
             },
         }
         layer_label = "livepatch"
+        force_restart = self._update_trusted_ca_certs(workload_container)
         workload_container.add_layer(layer_label, update_config_environment_layer, combine=True)
+        self._restart_service(force_restart)
+
+    def _restart_service(self, force_restart):
```

Comment:
What about passing the container to the function, we already have the container and have checked we can connect to it.

Reply:
Also, restart_service isn't accurate as we either restart or start the service.

Reply:
renamed to _start_or_restart_service